### PR TITLE
Add `DO NOT EDIT` warn to bootstrap sync manifests

### PR DESCRIPTION
**Pull request by stefanprodan**
Fix: #2049

### CODE REVIEW

**Review by hiddeco — COMMENTED** [review]
Should we maybe provide a hint and link people to the customize documentation?

File: pkg/manifestgen/manifest.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	securejoin "github.com/cyphar/filepath-securejoin"
 )
 
+const GenWarning = "# This manifest was generated by flux bootstrap. DO NOT EDIT."
+
```
Comment (hiddeco) [inline pkg/manifestgen/manifest.go:28]:
```suggestion
const GenWarning = `# This manifest was generated by flux bootstrap. DO NOT EDIT.
# For modifications to the Flux configuration, see the "customize Flux" documentation.
# Ref: https://fluxcd.io/docs/installation/#customize-flux-manifests
`

```

Comment (stefanprodan) [reply pkg/manifestgen/manifest.go:28]: What about Terraform? this comment ends up everywhere manifestgen is used

Comment (stefanprodan) [reply pkg/manifestgen/manifest.go:28]: I think we need to have a dedicated page for "Customize Flux manifests" where we address both the CLI and TF bootstrap options. After that we could update this comment and point to the new page.

Comment (hiddeco) [reply pkg/manifestgen/manifest.go:28]: We should record this in a follow-up issue.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**